### PR TITLE
Development of register view

### PR DESCRIPTION
**Pull request by Hanya-Ahmad**
### Description:
Developed register view with its validations and modified input labels' styling across all views.
### Issue
- #39 
### Register View Design
1. Landing page
image
2. Invalid username
image
3. Empty password 
image
4. Password does not contain numbers
image
5. Password does not contain special characters
image
6. Passwords do not match
image
7. Terms and privacy policy not accepted
image
8- Valid form
image
 ### Login and Forget Password Views Updated Design
image
image

### CODE REVIEW

File: frontend/src/components/auth/RegisterComponent.vue [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+            <v-checkbox type="checkbox" hide-details v-model="termsCheck">
+              <template v-slot:label>
+                <span class="terms-checkbox"
+                  >I agree to all the <a id="terms-link">Terms</a> and
+                  <a id="privacy-policy-link">Privacy policy.</a></span
+                >
+              </template>
+            </v-checkbox>
+            <v-btn
+              type="submit"
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/RegisterComponent.vue]:
```suggestion
              type="Register"
```
or 
```suggestion
              type="Sign up"
```

File: frontend/src/utils/validations.ts [hunk 2]
```diff
@@ -26,21 +26,80 @@ export function validateEmail(value: string): ValidationResult {
...
 
+//isEightCharsLong checks if password is at least eight characters
+function isEightCharsLong(value: string): boolean {
+  if (value.length >= 8) return true;
+  return false;
+}
+
+//containsOneLetter checks if password contains at least one letter
+function containsOneLetter(value: string): boolean {
+  const regexPattern = /[A-Za-z]/;
```
Comment (mahmoud-emad) [inline frontend/src/utils/validations.ts]: Let's move this content to the top of he file and rename it to `R_CONTAIN_ONE_LETTER`

File: frontend/src/utils/validations.ts [hunk 3]
```diff
@@ -26,21 +26,80 @@ export function validateEmail(value: string): ValidationResult {
...
+  return regexPattern.test(value);
+}
+
+//containsOneNumber checks if password contains at least one number
+function containsOneNumber(value: string): boolean {
+  const regexPattern = /\d/;
```
Comment (mahmoud-emad) [inline frontend/src/utils/validations.ts]: Same here

File: frontend/src/utils/validations.ts [hunk 4]
```diff
@@ -26,21 +26,80 @@ export function validateEmail(value: string): ValidationResult {
...
+  return regexPattern.test(value);
+}
+
+//containsOneSymbol checks if password contains at least one special character
+function containsOneSymbol(value: string): boolean {
+  const regexPattern = /[_@$!%*#?&]/;
```
Comment (mahmoud-emad) [inline frontend/src/utils/validations.ts]: Same here

File: frontend/src/views/auth/RegisterView.vue [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  </div>
+</template>
+
+<script lang="ts">
+import LogoImage from "../../components/LogoImage.vue";
+import RegisterComponent from "../../components/auth/RegisterComponent.vue";
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  name: "LoginPage",
```
Comment (mahmoud-emad) [inline frontend/src/views/auth/RegisterView.vue]: LoginPage??

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]
Excellent work! 😄 I've taken the liberty of adding some comments that I believe could enhance both the code quality and performance. Thank you for your dedication!

**Comment by Mahmoud-Emad** [comment]
One more comment, It looks like it something hidden

- image

- image

- image

**Review by Mahmoud-Emad — APPROVED** [review]

**Pull request merged**